frontend/src/components/AbstractModal.tsx

```diff
@@ -42,34 +40,34 @@ const useStyles = makeStyles((theme) => ({
   },
 }))
 
-export default function AbstractModal(props: ModalType): JSX.Element {
+export default function AbstractModal({
+  title,
+  description,
+  confirmText,
+  cancelText,
+  isModalOpen,
+  handleConfirm,
+  handleCancel,
+}: ModalType): JSX.Element {
   const classes = useStyles()
 
   return (
-    <Dialog open={props.isOpen} classes={{ paper: classes.modal }}>
+    <Dialog open={isModalOpen} classes={{ paper: classes.modal }}>
       <DialogTitle>
-        <IconButton className={classes.modalTitle} onClick={props.handleCancel}>
+        <IconButton className={classes.modalTitle} onClick={handleCancel}>
           <CloseIcon />
         </IconButton>
       </DialogTitle>
       <DialogContent className={classes.modalContent}>
-        <Typography variant="h6">{props.title}</Typography>
-        <Typography variant="subtitle2">{props.description}</Typography>
+        <Typography variant="h6">{title}</Typography>
+        <Typography variant="subtitle2">{description}</Typography>
       </DialogContent>
       <DialogActions className={classes.modalAction}>
-        <Button
-          variant="contained"
-          onClick={props.handleCancel}
-          color="secondary"
-        >
-          {props.cancelText || "NO"}
+        <Button variant="contained" onClick={handleCancel} color="secondary">
+          {cancelText || "NO"}
```

Comment:
If I'm being picky, this could be the default value of `cancelText`, since the parameter is optional

Reply:
yeah you're right